### PR TITLE
Update deprecation sections to reflect new policy

### PR DESCRIPTION
**Pull request by WardBrian**
#### Submission Checklist

- [x] Builds locally 
- [x] New functions marked with `` `r since("VERSION")` ``
- [x] Declare copyright holder and open-source license: see below

#### Summary

Update deprecation sections to reflect https://github.com/stan-dev/design-docs/pull/43

Related to https://github.com/stan-dev/stanc3/pull/1044

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting (this will be you or your assignee, such as a university or company):

Simons Foundation

By submitting this pull request, the copyright holder is agreeing to license the submitted work under the following licenses:
- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: src/reference-manual/deprecations.Rmd
Comment (bob-carpenter) [inline src/reference-manual/deprecations.Rmd:8]: same question again

File: src/reference-manual/deprecations.Rmd
Comment (bob-carpenter) [inline src/reference-manual/deprecations.Rmd:13]: There's always a comma after "e.g.", as in `(e.g., Stan 3.0)`.

File: src/reference-manual/deprecations.Rmd
Comment (bob-carpenter) [inline src/reference-manual/deprecations.Rmd:132]: Nice to see the attention to boundary conditions!

File: src/reference-manual/deprecations.Rmd
Comment (bob-carpenter) [inline src/reference-manual/deprecations.Rmd:136]: There's a drop-in replacement for this.  I think it's just a lognormal prior on the scales with an LKJ prior on the correlations.  It's not a one-symbol for one-symbol replacement, though.  So probably not something that can be auto-handled by the pretty printer.

**Review by bob-carpenter — CHANGES_REQUESTED** [review]
thanks---just a few minor comments.

File: src/functions-reference/deprecated_functions.Rmd
Comment (WardBrian) [reply src/functions-reference/deprecated_functions.Rmd:9]: Yes, this is an RMarkdown special case where if you want to link to a section header and the link text is the same as the header, you can omit the (). [documentation on it](https://bookdown.org/yihui/bookdown/cross-references.html)

File: src/reference-manual/deprecations.Rmd
Comment (WardBrian) [reply src/reference-manual/deprecations.Rmd:136]: I agree, it's just barely out-of-scope in my opinion

**Review by bob-carpenter — APPROVED** [review]
Thanks.

**Comment by bob-carpenter** [comment]
It's still marked as draft, but if your intention is to merge, you should be able to do that now.

**Comment by WardBrian** [comment]
@bob-carpenter I'm going to see if any further documentation needs come up in the next couple stanc PRs related to this before merging, but if nothing changes I'll merge after those.

**Pull request merged**
